src/features/dashboard/components/ApiTokenCard/index.tsx

```diff
@@ -15,25 +15,33 @@ interface IApiTokenCardProps {
   description: string;
 }
 
-const ApiTokenCard = ({ register, name, label, description }: IApiTokenCardProps) => {
+const ApiTokenCard = ({ register, name, label, description, ...rest }: IApiTokenCardProps) => {
   const [isAdminChecked, setIsAdminChecked] = useState(false);
   const [isAdminPopupVisible, setIsAdminPopupVisible] = useState(false);
   const { deviceType } = useDeviceType();
 
-  const handleAdminScopeChange = (e?: React.ChangeEvent<HTMLInputElement>, chk?: boolean) => {
-    if (e) {
-      const isChecked = e.target.checked;
-      setIsAdminChecked(isChecked);
-      setIsAdminPopupVisible(isChecked);
-    } else if (chk) {
-      setIsAdminPopupVisible(false);
+  const handleAdminScopeChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
can we refactor this function as I see lot of redundancy of setState's in `handleAdminScopeChange, handleModalPrimaryButton and handleModalSecondaryButton` ?